pkg/controller/datavolume/util.go

```diff
@@ -380,22 +381,22 @@ func getDefaultVolumeAndAccessMode(c client.Client, storageClass *storagev1.Stor
 	return nil, nil, errors.Errorf("no accessMode specified in StorageProfile %s", storageProfile.Name)
 }
 
+// Get the StorageProfile preferred VolumeMode based on the StorageClass and AccessModes
 func getDefaultVolumeMode(c client.Client, storageClass *storagev1.StorageClass, pvcAccessModes []v1.PersistentVolumeAccessMode) (*v1.PersistentVolumeMode, error) {
 	if storageClass == nil {
 		// fallback to k8s defaults
 		return nil, nil
 	}
+	if len(pvcAccessModes) == 0 {
+		return nil, errors.Errorf("no accessModes specified")
```

Comment:
Is even possible to reach this condition? If both `pvcAccessModes` and `VolumeMode` are empty we would be in `getDefaultVolumeAndAccessMode` instead, right?

Reply:
right, so we can drop this check and just in case someone somehow passes empty `pvcAccessModes`, we'll return `nil, nil` as we already do if no match. wdyt?